notebooks/03-query/07-functions-datetime.ipynb

```diff
@@ -0,0 +1,944 @@
+    "  COUNT(*) AS \"sessions\"\n",
+    "FROM \"example-koalas-fndatetime\"\n",
+    "WHERE TIME_IN_INTERVAL(__time,'2019-08-25T11/2019-08-25T13')\n",
+    "'''\n",
+    "\n",
+    "display.sql(sql)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a5f9c823-50bb-4f1d-b0a1-78a848641c93",
+   "metadata": {},
+   "source": [
+    "The second parameter is an [ISO8601](https://en.wikipedia.org/wiki/ISO_8601#Time_intervals) interval representing the period between 11am and 1pm on the 25th August 2019.\n",
+    "\n",
+    "In the next cell, an ISO8601 period is given as the start of the interval to give the same results."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "5a5ed8e2-c33f-447f-8bed-4f1dca034f0c",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sql='''\n",
+    "SELECT\n",
+    "  MIN(__time) AS \"earliestSession\",\n",
+    "  MAX(__time) AS \"latestSession\",\n",
+    "  COUNT(*) AS \"sessions\"\n",
+    "FROM \"example-koalas-fndatetime\"\n",
+    "WHERE TIME_IN_INTERVAL(__time,'PT2H/2019-08-25T13')\n",
+    "'''\n",
+    "\n",
+    "display.sql(sql)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e8765ecd-35ff-41bf-810b-913fbc8f408e",
+   "metadata": {},
+   "source": [
+    "The `TIME_SHIFT` function applies multiples of ISO8601 periods to a timestamp.\n",
+    "\n",
+    "Run the next cell for a simple example:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "0df48f7f-5fcf-4ecd-80c9-3e0d9dca1b72",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sql='''\n",
+    "SELECT\n",
+    "  __time AS \"sessionStart\",\n",
+    "  TIME_SHIFT(__time,'PT1H',3) AS \"sessionStart-plus3Hours\",\n",
+    "  TIME_SHIFT(__time,'P1Y',-5) AS \"sessionStart-minus5Years\"\n",
+    "FROM \"example-koalas-fndatetime\"\n",
+    "WHERE TIME_IN_INTERVAL(__time,'2019-08-25T13/PT1S')\n",
+    "'''\n",
+    "\n",
+    "display.sql(sql)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "714dca10-f080-4341-9eda-e20775c20db1",
+   "metadata": {},
+   "source": [
+    "A period - along with a multiple - has been provided to the `TIME_SHIFT` function. This is then added to the timestamp. One calculation adds 3 hours, the next subtracts 5 years."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6b59657c-28e8-42c2-b387-65e0ed57c046",
+   "metadata": {},
+   "source": [
+    "The results are all sessions that started in the minute before 1pm and ended (by calculation) in the minute after 1pm.\n",
+    "\n",
+    "Run the following cell to see `TIME_SHIFT` and `TIME_IN_INTERVAL` being used together."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "87dd8e91-569d-4ad9-8a62-d839264d0484",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sql='''\n",
+    "SELECT\n",
+    "  __time AS \"start\",\n",
+    "  TIME_SHIFT(__time, 'PT0.01S', \"session_length\") AS \"end\"\n",
```

Comment:
adding the time shift as another column helps explain it, something like:
```
SELECT
  __time AS "start", 
  "session_length" * 0.01 as  session_seconds,
  TIME_SHIFT(__time, 'PT0.01S', "session_length") AS "end"
FROM "example-koalas-fndatetime"
WHERE TIME_IN_INTERVAL(TIME_SHIFT(__time, 'PT0.01S', "session_length"),'2019-08-25T13/PT1M')
AND TIME_IN_INTERVAL(__time, 'PT1M/2019-08-25T13')
LIMIT 10
```

Reply:
Added - thank you!